pkg/apis/kf/v1alpha1/app_masks.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+// AppSpecSourceMask is a _shallow_ copy of the SourceSpec object to a new
+// SourceSpec object bringing over only the fields allowed to be set in
+// the App by developers. This does not validate the contents or the bounds of
+// the provided fields.
+func AppSpecSourceMask(in SourceSpec) SourceSpec {
```

Comment:
nit: can we add a comment suggesting this function is best used with https://godoc.org/knative.dev/pkg/apis#CheckDisallowedFields